src/en/RainOfSnow.lua

```diff
@@ -98,12 +115,23 @@ local function parseListing(listingURL)
 end
 
 local function getListing(data)
-	local docURL = expandURL("/novels/page/" ..data[PAGE])
-	return parseListing(docURL)
+	local page = data[PAGE] + 1
+	local orderBy = data[ORDER_BY_FILTER]
+	local genre = data[GENRE_FILTER]
+	
+	local url
+	-- Genre filtering only works with popular order by.
+	if genre ~= 0 then
+		url = "/novels/" .. "?n_orderby=" .. GENRE_TERMS[genre+1]
+	else
+		url = "/"	.. ORDER_BY_TERMS[orderBy+1].. "/page/" ..page
+	end
+
+	return parseListing(expandURL(url))
 end
 
 local function getSearch(data)
-	local docURL = expandURL("?s=" ..data[QUERY])
+	local docURL = expandURL("?s=" ..encode(data[QUERY]))
```

Comment:
Might be a taste thing on my part, but I would move the expandURL into the parseListing. Is fine though.

Reply:
I think it's fine for this extension